### DISCUSSION TITLE
I cannot pass parameter by method of another class  "report.writeDeleteReports(guitars[num-1]);" does not work. I attached all of source code.

### DISCUSSION THREAD

**Discussion by rktndnjseh** [community/community#118831] [Programming Help]
### Select Topic Area

Question

### Body

import java.util.Objects;
import java.util.Scanner;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
class Guitar {
    private final int stockNumber;
    private final String name;
    private final String type;
    private final int string;
    private final String brand;
    private final int price;

     Guitar(int stockNumber, String name, String type, int string, String brand, int price) {
        this.stockNumber = stockNumber;
        this.name = name;
        this.type = type;
        this.string = string;
        this.brand = brand;
        this.price = price;
    }

     int getStockNumber() {
        return stockNumber;
    }

     String getBrand() {
        return brand;
    }

     int getString() {
        return string;
    }

     String getType() {
        return type;
    }

     String getName() {
        return name;
    }

     int getPrice() {
        return price;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj)
            return true;
        if (getClass() != obj.getClass())
            return false;
        Guitar guitar = (Guitar) obj;

        return string == guitar.string &&
                price == guitar.price &&
                Objects.equals(name, guitar.name) &&
                Objects.equals(type, guitar.type) &&
                Objects.equals(brand, guitar.brand);
    }
}
    class GuitarStock {
        private final Guitar[] guitars = new Guitar[10];
        private final InputManager input = new InputManager();
        private final StockReport report = new StockReport();
        GuitarStock(){
            initStocks();
        }
        private void initStocks() {
            guitars[0] = new Guitar(1,"랩스틸", "베이스", 4, "fender", 120000);
            guitars[1] = new Guitar(2,"하와이완", "베이스", 4, "gretsh", 10000);
            guitars[2] = new Guitar(3,"레조네이터", "일렉", 6, "라운드", 150000);
            guitars[3] = new Guitar(4,"채프먼스틱", "채프먼스틱", 10, "데모", 100000);
            guitars[4] = new Guitar(5,"피들스틱", "일렉", 6, "라운드", 150000);
            guitars[5] = new Guitar(6,"스카너", "베이스", 6, "잠시", 300000);
            guitars[6] = new Guitar(7,"레조네이터", "일렉", 6, "라운드", 150000);
            guitars[7] = new Guitar(8,"레조네이터", "일렉", 6, "라운드", 150000);
            guitars[8] = new Guitar(9,"채프먼스틱", "채프먼스틱", 10, "데모", 100000);
            guitars[9] = new Guitar(10,"티모", "일렉", 8, "프렛", 250000);
        }

        public void guitarSearch() {
            System.out.println("검색할 재고의 정보를 입력하세요.");
            String name = input.getInput("기타의 이름을 입력하세요");
            String type = input.getInput("기타의 종류를 입력하세요");
            int string = input.getIntInput("기타의 줄 수를 입력하세요");
            String brand = input.getInput("기타의 브랜드를 입력하세요");
            int price = input.getIntInput("가격이 얼마 이하의 기타인지");
            boolean found = false;
            System.out.println("검색한 재고:\n");
            for (Guitar guitar : guitars) {
                if (guitar != null && isMatch(guitar, name, type, string, brand, price)) {
                    System.out.println("재고번호: " + guitar.getStockNumber() + ". 이름: " + guitar.getName() +
                            ", 종류: " + guitar.getType() + ", 줄 수: " + guitar.getString() +
                            ", 브랜드: " + guitar.getBrand() + ", 가격: " + guitar.getPrice() + "원");
                    found = true;
                }
            }if(!found){
                System.out.println("재고에 해당하는 기타가 없습니다.");
            }
        }
        private boolean isMatch(Guitar guitar, String name, String type, int string, String brand, int price) {
            boolean valid = true;
            if (!name.isEmpty() && !guitar.getName().equals(name)) {
                valid = false;
            }
            if (!type.isEmpty() && !guitar.getType().equals(type)) {
                valid = false;
            }
            if (string != 0 && guitar.getString() != string) {
                valid = false;
            }
            if (!brand.isEmpty() && !guitar.getBrand().equals(brand)) {
                valid = false;
            }
            if (price != 0 && guitar.getPrice() > price) {
                valid = false;
            }
            return valid;
        }

       public void addStock() {
           System.out.println("추가하고 싶은 재고의 정보를 입력하세요");
           int num = 0;
           for (int i = 0; i < guitars.length; i++) {
               if (guitars[i] == null) {
                   num=i;
                }
           }
           guitars[num-1] = createGuitar(num);
           System.out.println(num+"번으로 기타가 추가되었습니다.");
           report.writeAddReports(guitars[num-1]);
       }
       public void sellStock(){
           System.out.println("판매한 재고의 번호를 입력하세요");
           int num = input.getIntInput("재고 번호를 모른다면 1을 입력하세요");
           if(num == 1){
               guitarSearch();
           }else if (num < 1 || num > 10 || guitars[num - 1] == null) {
               System.out.println("잘못된 입력입니다.");
               return;
           }
           guitars[num - 1] = null;
       }
        public void deleteStock() {
            guitarSearch();
            int num = input.getIntInput("삭제하고 싶은 기타의 재고번호를 선택하세요 (그만 입력하고 싶다면 -1을 입력하세요)");
            if (num == -1) return;
            if (num < 1 || num > 10 || guitars[num - 1] == null) {
                System.out.println("잘못된 입력입니다.");
                return;
            }
            report.writeDeleteReports(guitars[num-1]);
            guitars[num - 1] = null;
            System.out.println(num + "번 기타가 삭제되었습니다.");
        }
        public void editStock() {
            guitarSearch();
            int num = input.getIntInput("수정하고 싶은 기타의 재고번호를 선택하세요 (그만 입력하고 싶다면 -1을 입력하세요)");
            if (num == -1) return;
            if (num < 1 || num > 10 || guitars[num - 1] == null) {
                System.out.println("잘못된 입력입니다.");
                return;
            }
            Object beforeGuitar = guitars[num-1];
            guitars[num - 1] = createGuitar(num);
            System.out.println(num + "번 기타가 수정되었습니다.");
            report.writeEditReports(beforeGuitar, guitars[num-1]);
        }
        private Guitar createGuitar(int stockNumber) {
            String name = input.getInput("기타의 이름을 입력하세요");
            String type = input.getInput("기타의 종류를 입력하세요");
            int string = input.getIntInput("기타의 줄 수를 입력하세요");
            String brand = input.getInput("기타의 브랜드를 입력하세요");
            int price = input.getIntInput("기타의 가격을 입력하세요");
            return new Guitar(stockNumber, name, type, string, brand, price);
        }
    }
class CurrentTime{
    public String getCurrentTime() {
        LocalDateTime now = LocalDateTime.now();
        return now.format(DateTimeFormatter.ofPattern("yyyy년 MM월 dd일 HH시 "));
    }
}

    class  StockReport{
    private final String[] addReports ={};
    private final String[] deleteReports = new String[10];
    private final String[] editReports ={};
    private final CurrentTime currentTime = new CurrentTime();
    public void printAddReports(){
        for(int i=0;i<10;i++){
            if(addReports[i]!=null){
                System.out.println(addReports[i]);
            }
        }
    }
    public void printDeleteReports(){
        for(int i=0;i<10;i++){
            if(deleteReports[i]!=null){
                System.out.println(deleteReports[i]);
            }
        }
    }
    public void printEditReports(){
        for(int i=0;i<10;i++){
            if(editReports[i]!=null){
                System.out.println(editReports[i]);
            }
        }
    }
    public void writeAddReports(Guitar guitar){
        for(int i=0;i<10;i++){
            if(addReports[i]==null){
                addReports[i]=currentTime.getCurrentTime()+guitar.getStockNumber()+"번 재고, 이름: "+guitar.getName()+", 종류: "
                +guitar.getType()+", 줄 수: "+guitar.getString()+", 브랜드: "+guitar.getBrand()+
                        ", 가격: "+ guitar.getPrice()+"가 추가되었습니다.";
            }
        }
    }

    public void writeDeleteReports(Guitar guitar){
        for(int i=0;i<10;i++){
            if(deleteReports[i]==null){
                deleteReports[i]=currentTime.getCurrentTime()+guitar.getStockNumber()+"번 재고, 이름: "+guitar.getName()+", 종류: "
                +guitar.getType()+", 줄 수: "+guitar.getString()+", 브랜드: "+guitar.getBrand()+
                        ", 가격: "+ guitar.getPrice()+"가 삭제되었습니다.";
                }
            }
        }
        public void writeEditReports(Object beforeGuitar, Guitar guitar){
            for(int i=0;i<10;i++){
                if(editReports[i]==null){
                    Guitar before = (Guitar) beforeGuitar;
                    editReports[i]= currentTime.getCurrentTime()+
                            before.getStockNumber()+"번 재고, 이름: "+before.getName()+", 종류: "
                            +before.getType()+", 줄 수: "+before.getString()+", 브랜드: "+before.getBrand()+
                            ", 가격: "+ before.getPrice()+"가\n"+ guitar.getStockNumber()+
                            "번 재고, 이름: "+guitar.getName()+", 종류: "
                    +guitar.getType()+", 줄 수: "+guitar.getString()+", 브랜드: "+guitar.getBrand()+
                            ", 가격: "+ guitar.getPrice()+"가 수정되었습니다.";
                }
            }
        }
    }

    class InputManager {
        private final Scanner input = new Scanner(System.in);
        private final StockReport report = new StockReport();

        String getInput(String message) {
            System.out.println(message);
            return input.nextLine();
        }

        int getIntInput(String message) {
            System.out.println(message);
            String in = input.nextLine();
            return in.isEmpty() ? 0 : Integer.parseInt(in);
        }

        int getOrder() {
            System.out.println("초기메뉴:");
            System.out.println("1. 기타 검색, 2. 재고 추가, 3. 재고 수정, 4. 재고 삭제, 5. 판매, 6. 판매 보고서 출력, 6. 재고 보고서 출력");
            System.out.println("원하는 메뉴를 선택하세요");
            return input.nextInt();
        }

        void printStockReport() {
            System.out.println("추가된 재고에 대한 보고서를 원하시면 1번");
            System.out.println("수정된 재고에 대한 보고서를 원하시면 2번");
            System.out.println("삭제된 재고에 대한 보고서를 원하시면 3번을 선택하세요");
            System.out.println("그만 입력하고 싶으면 0번을 선택하세요");
            while (true) {
                int order = input.nextInt();
                if (order == 0) {
                    break;
                }
                switch (order) {
                    case 1:
                        report.printAddReports();
                        break;
                    case 2:
                        report.printEditReports();
                        break;
                    case 3:
                        report.printDeleteReports();
                        break;
                    default:
                        System.out.println("1, 2, 3 중에서 선택하세요");
                        break;
                }
            }
        }
    }

public class Main {
    public static void main(String[] args) {
        GuitarStock stock = new GuitarStock();
        InputManager input = new InputManager();

        while(true){
            int order;

            order = input.getOrder();

            switch (order){
                case 1 :
                    stock.guitarSearch();
                    break;
                case 2 :
                    stock.addStock();
                    break;
                case 3 :
                    stock.editStock();
                    break;
                case 4 :
                    stock.deleteStock();
                    break;
                case 5 :

                    break;
                case 6 :
                    input.printStockReport();
                default:
                    break;
            }
        }
    }
}

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @rktndnjseh!

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Discussions category is solely related to conversations around the [GitHub product Discussions](https://docs.github.com/en/discussions/collaborating-with-your-community-using-discussions/about-discussions). This question should be in the Programming Help category. I've gone ahead and moved it for you. Good luck!
